src/components/home/Home.tsx

```diff
@@ -1,28 +1,53 @@
 import "./Home.scss";
 
+import { AppDispatch, RootState } from "state/rootReducer";
 import { EXACT_TRANSACTION, LOGIN_PATH, SUBMISSIONS_PATH, TRANSACTIONS_PATH } from "routes/routes";
 import { Redirect, Route, RouteComponentProps, Switch, withRouter } from "react-router-dom";
+import { setLoggedIn, setLoggedOut, setLoggingIn } from "components/login/state/statusAction";
 
 import { Footer } from "components/footer/Footer";
 import Header from "components/header/Header";
 import { LoggedIn } from "components/logged-in/LoggedIn";
 import Login from "components/login/Login";
 import { LoginStatus } from "enums/loginStatus";
 import React from "react";
-import { RootState } from "state/rootReducer";
 import { SANDBOX_PATH } from "sandbox/Sandbox";
 import { connect } from "react-redux";
+import { handleAutoLogin } from "components/login/login-service/loginService";
 
 const mapStateToProps = (state: RootState) => ({
     logginStatus: state.loginState.status,
 });
 
-type ConnectedHomeProps = ReturnType<typeof mapStateToProps> & RouteComponentProps;
+const mapDispatchToProps = (dispatch: AppDispatch) => ({
+    setLoggingIn: () => dispatch(setLoggingIn()),
+    setLoggedIn: (idToken: string, accessToken: string) => dispatch(setLoggedIn(idToken, accessToken)),
+    setLoggedOut: () => dispatch(setLoggedOut()),
+});
+
+type ConnectedHomeProps = ReturnType<typeof mapStateToProps> &
+    ReturnType<typeof mapDispatchToProps> &
+    RouteComponentProps;
 
 class Home extends React.Component<ConnectedHomeProps> {
+    componentDidMount = () => {
+        handleAutoLogin(
+            this.props.location.pathname,
+            this.props.setLoggingIn,
+            this.props.setLoggedIn,
+            this.props.setLoggedOut,
+        );
+    };
+
+    componentDidUpdate = (oldProps: ConnectedHomeProps) => {
+        if (oldProps.logginStatus !== LoginStatus.loggedIn && this.props.logginStatus === LoginStatus.loggedIn) {

```

Comment:
I think sticking to the more common pattern of checking if it has changed would make this easier to work with in the future.
```suggestion
        if (oldProps.logginStatus !== this.props.logginStatus && this.props.logginStatus === LoginStatus.loggedIn) {

```